Feature updates

benfreke opened the pull request:
1. Don't output a TOC if there are no headings in a markdown file
2. Add a `grandchild` class to the output to allow better targeting via css
3. Gets the tests to run in a case sensitive environment (i.e. linux)
4. Update phpunit and benchmark to latest versions
5. Replaces a deprecated value in `CommonMark`

I have a `docker-compose` file that I tested with locally, which I have not included with this PR. I can easily do that if you'd like?

Also, since this would technically be a 1.1 release feel free to reject this if it is too large. I'd just need to fork this and use it locally if that is the case.

I've tried to use a TDD approach so you can follow along in the commits (if you're interested) to see what changed and why.

benfreke committed:
Adds php8 support

benfreke committed:
Ensure php73 is supported

benfreke committed:
Merge branch 'master' of github.com:benfreke/statamic-toc

benfreke committed:
Adds failing test

benfreke committed:
Fixes failing test

benfreke committed:
Makes the tests run in a Linux environment

Linux is case sensitive, and the directory path was wrong.
I've also upgraded phpunit and testbench to the latest version to remove warning about coverage

benfreke committed:
Fixes tests failing in PHP7.3

benfreke committed:
Removes unused values

benfreke committed:
Better name for the method

benfreke committed:
Updates deprecated value

benfreke committed:
Removes included class that is no longer used

benfreke committed:
Adds failing test for additional classes on list elements

benfreke committed:
Fixes test so that h4 classes are asserted for

benfreke committed:
Adds comments to help illustrate purpose of class

And remove the extra code that isn't needed

benfreke commented:
https://github.com/orchestral/testbench/issues/302

That issue seems to indicate something about parallel, but I can't see that in the Github action. 
I'll dig into this later

benfreke committed:
Sets lowest and highest dependencies so tests succeed

benfreke committed:
Adds more variables to the test matrix to cover eventualities

- Removes tests for php8 and lowest statamic version, as support is added in 3.0.34
- Adds a matrix key for statamic, as 3.1 wasn't being tested
- Attempts to set the correct strings for versions of statamic used
- Removes deprecated `no-suggest` flag of composer

benfreke committed:
Fixes workflow syntax error

benfreke commented:
This has added a lot of extra jobs, so i don't think this was the right way forward.

If you'd like @jedamzik , I can change this to be a 2.0 release, and drop Statamic 3.0, L7 and PHP 73 support? There are a lot of changes here that would make this a `1.1` release, so making removing support for older versions might be better for your Github minutes?

jedamzik commented:
@benfreke Thanks for the work you put in. I didn't get around to looking into the Statamic `3.1` release before now, so this helps a lot!

In my opinion, this should work just fine as a `1.1` release. Github Action minutes are free for public repos (https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) so no need to drop support for older versions just because of that 👍 

I'm guessing the testbench issues can be resolved by requiring `^6.17`: `"orchestra/testbench": "^v5.0 | ^v6.17",`, but I'll have to check on that.

benfreke committed:
Bumps testbench to a min of 6.17

This is only for the `--prefer-lowest` when using testbench v6

benfreke committed:
Sets min version for mockery

Copied from the Statamic workflow.
This should result in mockery 1.3.3 being installed which removes the isArray error
Also removes the testbench min requirement

benfreke committed:
Removes accidentally committed changes

benfreke committed:
Fixes typo in statamic version

benfreke committed:
Removes space that breaks workflow

benfreke committed:
Removes options not required for setting up tests

benfreke commented:
@jedamzik I haven't done a full check of the workflow actions, but this is now passing at least. 

It might be worthwhile adding an extra version of statamic to the tests (`3.0.*`) and exclude that from PHP 8 as well, but I think this has more than enough level of coverage. 

`v1.1` is good for me too. Thanks for creating this in the first place!
